### PR TITLE
initial patientsearch implementation

### PR DESCRIPTION
**Pull request by pbugni**
Ported the stayhome code to fit COSRI Patient Search.

Appears functional, forces keycloak login.  Routes the PDMP search through the backend flask client thus eliminating the CORS issues.  Timeouts and logout also working.

Pending for subsequent PR: handling the results from a PDMP patient search.

### CODE REVIEW

File: client_secrets.json.default [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-        "token_uri": "https://keycloak-dev.cirg.washington.edu/auth/realms/Stayhome/protocol/openid-connect/token",
-        "token_introspection_uri": "https://keycloak-dev.cirg.washington.edu/auth/realms/Stayhome/protocol/openid-connect/token/introspect"
+        "auth_uri": "https://keycloak-dev.cirg.washington.edu/auth/realms/cosri-launcher/protocol/openid-connect/auth",
+        "client_id": "cosri-patientsearch",
+        "client_secret": "ask-paul or check keycloak-dev",
+    	"issuer": "https://keycloak-dev.cirg.washington.edu/auth/realms/cosri-launcher",
+        "redirect_uris": ["http://localhost:8000/oidc_callback"],
+        "userinfo_uri": "https://keycloak-dev.cirg.washington.edu/auth/realms/cosri-launcher/protocol/openid-connect/userinfo",
+        "token_uri": "https://keycloak-dev.cirg.washington.edu/auth/realms/cosri-launcher/protocol/openid-connect/token",
+        "token_introspection_uri": "https://keycloak-dev.cirg.washington.edu/auth/realms/cosri-launcher/protocol/openid-connect/token/introspect"
```
Comment (ivan-c) [inline client_secrets.json.default:10]: Would be better to read these from env vars for consistent configuration

File: dashboard/src/js/components/Header.js [hunk 2]
```diff
@@ -1,23 +0,0 @@
-import React, { Component } from 'react';
```
Comment (ivan-c) [inline dashboard/src/js/components/Header.js:1]: Would we eventually be able to separate these components into a CIRG-wide library? just curious...

File: patientsearch.env.default [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # docker-compose default environment file
-# copy to dashboard.env and modify as necessary
+# copy to patientsearch.env and modify as necessary
 
+EXTERNAL_SEARCH_URL=https://cosri-pdmp.cirg.washington.edu/v/r2/fhir/
```
Comment (ivan-c) [inline patientsearch.env.default:4]:
```suggestion
FHIR_BASE_URL=https://cosri-pdmp.cirg.washington.edu/v/r2/fhir/
```
Not strictly specific to search, could extend to call other FHIR APIs

File: webpack.config.js [hunk 4]
```diff
@@ -3,20 +3,43 @@ const path = require('path');
...
+
+  if (env) {
+    // reduce it to a object
+    try {
+      envKeys = Object.keys(env).reduce((prev, next) => {
+        prev[`process.env.${next}`] = JSON.stringify(env[next]);
+        return prev;
+      }, {});
+    } catch(e) {
+      console.log("Error occurred processing .env config ", e);
```
Comment (ivan-c) [inline webpack.config.js:35]: I couldn't find a `.env` file in the repo, but wanted to caution that docker-compose also uses a `.env` file and can lead to some confusing results

Comment (achen2401) [reply webpack.config.js]: thank you, ivan. I did just remove this code

File: dashboard/src/js/components/Header.js (hunk 2 above)
Comment (achen2401) [reply dashboard/src/js/components/Header.js:1]: yeah, that will be a great direction.  something like this maybe:  https://bit.dev/frontend-teams

File: client_secrets.json.default (hunk 1 above)
Comment (pbugni) [reply client_secrets.json.default:10]: i agree - but request we make that a TODO, as the flask-oidc lib doesn't document how to easily move out of this magic filename.

File: patientsearch.env.default (hunk 3 above)
Comment (pbugni) [reply patientsearch.env.default:4]: Correct, but we need to somehow clarify "external" (that is PDMP in this case) versus the fEMR internal HAPI store, which FHIR_BASE_URL sounds likely to represent.

File: webpack.config.js (hunk 4 above)
Comment (pbugni) [reply webpack.config.js]: we got rid of that.  accidental oversight.

**Pull request merged**